rfq/manager.go

```diff
@@ -0,0 +1,594 @@
+		}
+
+		// Start the manager's main event loop in a separate goroutine.
+		m.Wg.Add(1)
+		go func() {
+			defer func() {
+				m.Wg.Done()
+
+				// Attempt to stop all subsystems if the main
+				// event loop exits.
+				err = m.stopSubsystems()
+				if err != nil {
+					log.Errorf("Error stopping RFQ "+
+						"subsystems: %v", err)
+				}
+
+				// The context can now be cancelled as all
+				// dependant components have been stopped.
+				cancel()
+			}()
+
+			log.Info("Starting RFQ manager main event loop")
+			m.mainEventLoop()
+		}()
+	})
+	return startErr
+}
+
+// Stop attempts to stop the RFQ manager.
+func (m *Manager) Stop() error {
+	var stopErr error
+
+	m.stopOnce.Do(func() {
+		log.Info("Stopping RFQ system")
+		stopErr = m.stopSubsystems()
+
+		// Stop the main event loop.
+		close(m.Quit)
+	})
+
+	return stopErr
+}
+
+// stopSubsystems stops the RFQ subsystems.
+func (m *Manager) stopSubsystems() error {
+	// Stop the RFQ order handler.
+	err := m.orderHandler.Stop()
+	if err != nil {
+		return fmt.Errorf("error stopping RFQ order handler: %w", err)
+	}
+
+	// Stop the RFQ stream handler.
+	err = m.streamHandler.Stop()
+	if err != nil {
+		return fmt.Errorf("error stopping RFQ stream handler: %w", err)
+	}
+
+	// Stop the RFQ quote negotiator.
+	err = m.negotiator.Stop()
+	if err != nil {
+		return fmt.Errorf("error stopping RFQ quote negotiator: %w",
+			err)
+	}
+
+	return nil
+}
+
+// handleIncomingMessage handles an incoming message. These are messages that
+// have been received from a peer.
+func (m *Manager) handleIncomingMessage(incomingMsg rfqmsg.IncomingMsg) error {
+	// Perform type specific handling of the incoming message.
+	switch msg := incomingMsg.(type) {
+	case *rfqmsg.Request:
+		err := m.negotiator.HandleIncomingQuoteRequest(*msg)
+		if err != nil {
+			return fmt.Errorf("error handling incoming quote "+
+				"request: %w", err)
+		}
+
+	case *rfqmsg.Accept:
+		// TODO(ffranr): The stream handler should ensure that the
+		//  accept message corresponds to a request.
+		//
+		// The quote request has been accepted. Store accepted quote
+		// so that it can be used to send a payment by our lightning
+		// node.
+		m.peerAcceptedQuotesMtx.Lock()
+		defer m.peerAcceptedQuotesMtx.Unlock()
+
+		scid := SerialisedScid(msg.ShortChannelId())
+		m.peerAcceptedQuotes[scid] = *msg
+
+		// Notify subscribers of the incoming quote accept.
+		event := NewIncomingAcceptQuoteEvent(msg)
+		m.publishSubscriberEvent(event)
+
+	case *rfqmsg.Reject:
+		// The quote request has been rejected. Notify subscribers of
+		// the rejection.
+		event := NewIncomingRejectQuoteEvent(msg)
+		m.publishSubscriberEvent(event)
+	}
+
+	return nil
+}
+
+// handleOutgoingMessage handles an outgoing message. Outgoing messages are
+// messages that will be sent to a peer.
+func (m *Manager) handleOutgoingMessage(outgoingMsg rfqmsg.OutgoingMsg) error {
+	// Perform type specific handling of the outgoing message.
+	switch msg := outgoingMsg.(type) {
```

Comment:
nit: should be an if, or will be expanded later?

Reply:
Will almost certainly be extended later.